script.js

```diff
@@ -1,12 +1,355 @@
+
+  popUp.innerHTML = `
+  <button id="popupCloseBtn">X</button>
+  <h2>${episodeData.name} - ${pad(episodeData)}</h2>
+  <div class="image">
+    <img src=${episodeData.image.medium} alt='${episodeData.season} ${episodeData.name
+    }'/>
+  </div>
+  <div class="description">
+    ${episodeData.summary}
+  </div>
+`;
+
+  // button for closing the popUp:
+  //get the button and add event listener
+  let closeBtn = popUp.children[0];
+  closeBtn.addEventListener("click", (e) => {
+    // remove the popup from the DOM
+    popUp.remove();
+  });
+
+  return popUp;
+}
+
+
+function makePageLayout() {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  //header
+  let header = createHeader();
+  rootElem.appendChild(header);
+  //section
+  let section = createSection();
+  // section will contain episodes, so it is hidden at the beginning
+  section.style.display = 'none';
+  rootElem.appendChild(section);
+  let selectE = document.querySelector(".select");
+  selectE.style.display = 'none';
+  // the popup will show the chosen episode.
+  // when selected value changes, a popup with the episode will open
+  selectE.addEventListener("change", (e) => {
+    let episodeId = e.target.value;
+    const popUp = createPopUp(episodeId);
+    rootElem.appendChild(popUp);
+  });
+
+
+}
+
+function makeHomeBtn() {
+  let section = document.querySelector('.section')
+  let showView = document.querySelector('.showView')
+
+  if (section.style.display == 'none') {
+    section.style.display = '';
+  } else {
+    section.style.display = 'none';
+  }
+
+  if (showView.style.display == 'none') {
+    showView.style.display = '';
+  } else {
+    showView.style.display = 'none';
+
+  }
+}
+
+function hideShowView() {
+  let section = document.querySelector('.section')
+  let showView = document.querySelector('.showView')
+  section.style.display = '';
+
+  showView.style.display = 'none';
+
+}
+
+//to display page
+function makePageForEpisodes(episodeList) {
+  //search episodes
+  let counter = document.getElementById("count");
+  let input = document.querySelector(".search");
+  input.addEventListener("keyup", (event) => {
+
+    const searchString = event.target.value.toLowerCase();
+    const episodeDivs = document.querySelectorAll(".episodeItem");
+    let countEpisodes = 0;
+
+
+    // Checks if each div contains the search string and if not sets display to none
+    for (let i = 0; i < episodeDivs.length; i++) {
+      if (!episodeDivs[i].innerHTML.toLowerCase().includes(searchString)) {
+        episodeDivs[i].style.display = "none";
+      } else {
+        episodeDivs[i].style.display = "block";
+        countEpisodes++;
+      }
+    }
+    counter.innerText = `Displaying   ${countEpisodes}/72 episode's`;
+  });
+  //end of search episodes
+  let section = document.querySelector('.section');
+  //display episodes
+  // remove old episodes
+  while (section.children.length > 0) {
```

Comment:
Great understanding of basic javascript functionality, conditionals and loops and using these tools to return the desired requirements. Keep on practising and getting familiar with these basic principles, they are the building blocks to amazing programs, apps and more. Line 239